src/MICore/Transports/PipeTransport.cs

```diff
@@ -35,6 +36,14 @@ protected override string GetThreadName()
             return "MI.PipeTransport";
         }
 
+        public override int DebuggerPid
+        {
+            get
+            {
+                return _debuggerPid;
```

Comment:
For pipe transport we are using the pid of the pipe process and for TCP transport I think we are throwing NotImplementedException. It seems like these two should be consistent with each other. I might suggest either making this an int? and returning null in both cases, or using 0 as an invalid value and returning that in both cases.